src/services/pcn-packetcapture/src/cbpf2c.cpp

```diff
@@ -0,0 +1,507 @@
+                              snprintf(ret, sizeof(ret),"L%d:\t//(%s)\n\tif((data + %s) > data_end){\n\t\treturn RX_DROP;\n\t}\n\ta = * ((uint8_t *) &data[%s]);\n",n, op, operand2, operand);
+                            } else {
+                                if ((bpf.code) == (BPF_LDX_B | BPF_MSH)){
+                                  char operand2[64];
+                                  snprintf(operand2, sizeof(operand), fmt, v + 1);
+                                  operand2[sizeof(operand2) - 1] = '\0';
+                                  snprintf(ret, sizeof(ret),"L%d:\t//(%s)\n\tif((data + %s) > data_end){\n\t\treturn RX_DROP;\n\t}\n\tx = * ((uint8_t *) &data[%s]);\n",n, op, operand2, operand);
+                                } else {
+                                    if ((bpf.code) == (BPF_LD_H | BPF_ABS)) {
+                                      char operand2[64];
+                                      snprintf(operand2, sizeof(operand), fmt, v + 2);
+                                      operand2[sizeof(operand2) - 1] = '\0';
+                                      snprintf(ret, sizeof(ret),"L%d:\t//(%s)\n\tif((data + %s) > data_end){\n\t\treturn RX_DROP;\n\t}\n\ta = ntohs(* ((uint16_t *) &data[%s]));\n",n, op, operand2, operand);
+                                    } else {
+                                        if (BPF_CLASS(bpf.code) == BPF_LD) {
+                                          char operand2[64];
+                                          snprintf(operand2, sizeof(operand), fmt, v + 4);
+                                          operand2[sizeof(operand2) - 1] = '\0';
+                                          snprintf(ret, sizeof(ret),"L%d:\t//(%s)\n\tif((data + %s) > data_end){\n\t\treturn RX_DROP;\n\t}\n\ta = ntohl(* ((uint32_t *) &data[%s]));\n",n, op, operand2, operand);
+                                        } else {
+                                            if (BPF_CLASS(bpf.code) == BPF_LDX) {
+                                              char operand2[64];
+                                              snprintf(operand2, sizeof(operand), fmt, v + 4);
+                                              operand2[sizeof(operand2) - 1] = '\0';
+                                              snprintf(ret, sizeof(ret),"L%d:\t//(%s)\n\tif((data + %s) > data_end){\n\t\treturn RX_DROP;\n\t}\n\tx = ntohl(* ((uint32_t *) &data[%s]));\n",n, op, operand2, operand);
+                                            } else {
+                                              snprintf(ret, sizeof(ret), "L%d:\t%s;\n", n, operand);
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    ret[sizeof(ret) - 1] = '\0';
+    return ret;
+}
+
+const char *cbpf2c::cbpf_dump_linux_k(uint32_t k) {
```

Comment:
Can you please add a comment about this function?

Reply:
done